### PR TITLE
Controlled pool participation

### PR DESCRIPTION
**Pull request by lukasz-zimnoch**
This changeset changes the rules of joining and leaving the pool by operators. From now, the contract owner controls those actions and operators should route their joining requests through. The `removeOperator` action should allow the owner to kick out operators in an arbitrary way. 

Changes provided in this PR will be used in the random beacon v2 to kick out operators who missed their turn for relay entry submission. The `RandomBeacon` contract will own the sortition pool and control join/leave actions. See: https://github.com/keep-network/keep-core/pull/2664 and https://github.com/keep-network/keep-core/pull/2671

### CODE REVIEW

File: contracts/SortitionPool.sol [hunk 1]
```diff
@@ -39,15 +39,26 @@ contract SortitionPool is SortitionTree {
...
+
     uint256 eligibleWeight = getEligibleWeight(operator);
     require(eligibleWeight > 0, "Operator not eligible");
 
     insertOperator(operator, eligibleWeight);
   }
 
+  /// @notice Removes an operator from the pool.
+  /// @dev Can be called only by the contract owner.
+  function leavePool(address operator) public {
```
[the remark this answers is not in the document]
Comment (pdyraga) [reply contracts/SortitionPool.sol]: Please also see #126.

[the remark this answers is not in the document]
Comment (lukasz-zimnoch) [reply contracts/SortitionPool.sol]: Addressed in: https://github.com/keep-network/sortition-pools/pull/125/commits/468ae1082e58856a265e0b5c9a74f2b80a064d5c. BTW, changed `updateOperatorStatus` to accept operator ID as well.

File: contracts/SortitionPool.sol [hunk 2]
```diff
@@ -39,25 +39,57 @@ contract SortitionPool is SortitionTree {
     );
   }
 
-  /// @notice Add an operator to the pool,
+  /// @notice Inserts an operator to the pool,
   /// reverting if the operator is already present.
-  function joinPool(address operator) public {
+  /// @dev Can be called only by the contract owner.
+  /// @param operator Address of the inserted operator.
+  function insertOperator(address operator) public {
```
Comment (pdyraga) [inline contracts/SortitionPool.sol:46]: This function could return operator ID. Non-blocking though, we can add it later once it's really needed.

File: test/sortitionPoolTest.js [hunk 3]
```diff
@@ -39,15 +39,111 @@ describe("SortitionPool", () => {
...
+
+      context("when operator is not eligible", () => {
+        it("should revert", async () => {
+          await expect(
+            pool.connect(owner).insertOperator(alice.address),
+          ).to.be.revertedWith("Operator not eligible")
+        })
+      })
+    })
+    context("when called by a non-owner", () => {
```
Comment (pdyraga) [inline test/sortitionPoolTest.js:63]: I'd expect linter to complain about missing blank line 🤔

**Review by pdyraga — APPROVED** [review]

**Pull request merged**